docs/pages/pmd/userdocs/cli_reference.md

```diff
@@ -87,26 +97,34 @@ The tool comes with a rather extensive help text, simply running with `--help`!
                             the given language `&lt;lang&gt;`. Parsing errors are ignored and unparsable files
                             are skipped.
                             
+                            <p>Use `--use-version` to specify the language version to use, if it is not the default.</p>
+
                             <p>This option allows to use the xml language for files, that don't
                             use xml as extension. See [example](#analyze-other-xml-formats) below.</p>"
     %}
     {% include custom/cli_option_row.html options="--ignore-list"
                option_arg="filepath"
                description="Path to file containing a list of files to ignore, one path per line.
-                            This option can be combined with `--dir` and `--file-list`.
-                            This ignore list takes precedence over any files in the file-list."
+                            This option overrides files included by any of `--dir`, `--file-list` and `--uri`."
     %}
-    {% include custom/cli_option_row.html options="--help,-h,-H"
+    {% include custom/cli_option_row.html options="--help,-h"
                description="Display help on usage."
     %}
-    {% include custom/cli_option_row.html options="-language,-l"
-               option_arg="lang"
-               description="Specify the language PMD should use. Used together with `-version`. See also [Supported Languages](#supported-languages)."
+    {% include custom/cli_option_row.html options="--use-version"
+               option_arg="lang-version"
+               description="The language version PMD should use when parsing source code.
+                            <p>Values are in the format of *language-version*.</p>
+                            <p>This option can be repeated to configure multiple languages to be analyzed during a single run</p>
+                            <p>See also [Supported Languages](#supported-languages).
+                            Using `--help` will display a full list of supported languages and versions.</p>"
     %}
-    {% include custom/cli_option_row.html options="--minimum-priority,-min"
-               option_arg="num"
-               description="Rule priority threshold; rules with lower priority than configured here won't be used."
-               default="5"
+    {% include custom/cli_option_row.html options="--minimum-priority"
+               option_arg="priority"
+               description="Rule priority threshold; rules with lower priority
+                              than configured here won't be used.
+                            Valid values (case insensitive): High, Medium High,
```

Comment:
I know this is out of scope, but I've always wanted to rename these versions to `higher, high, medium, lower, low`, to remove these awkward `medium high/low`... Wdyt?

Reply:
yeah, I agree the working is unclear, and actually bash completion for options with spaces is broken in PicoCli. I submitted https://github.com/remkop/picocli/pull/1759 for that, but it is still waiting

Reply:
+1 to reducing to 3. Specially since everything by default is medium. It's already rare for an org to manually override the value in their ruleset, they probably don't need so much granularity.

Additionally, it could be interesting to add an orthogonal value: confidence (high / low). Ie: when [`UnnecessaryImport`](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#unnecessaryimport) falls back to "best-effort" due to an incomplete auxclasspath, it could report violations at a lower confidence level.